src/main/java/kitchenpos/application/MenuService.java

```diff
@@ -40,53 +41,69 @@ public MenuService(
 
     @Transactional
     public Menu create(final MenuCreateRequest menuCreateRequest) {
-        final Long price = menuCreateRequest.getPrice();
+        MenuProducts menuProducts = generateMenuProducts(menuCreateRequest);
+        Menu menu = menuDao.save(generateMenu(menuCreateRequest, menuProducts));
 
-        if (Objects.isNull(price) || price < 0) {
-            throw new IllegalArgumentException();
+        final Long menuId = menu.getId();
+        List<MenuProduct> results = menuProducts.changeMenuId(menuId).getMenuProducts();
+        results.forEach(menuProductDao::save);
+        return menu.changeMenuProducts(results);
+    }
+
+    public List<Menu> list() {
+        final List<Menu> menus = menuDao.findAll();
+
+        for (final Menu menu : menus) {
+            menu.setMenuProducts(menuProductDao.findAllByMenuId(menu.getId()));
         }
 
-        if (!menuGroupDao.existsById(menuCreateRequest.getMenuGroupId())) {
+        return menus;
+    }
+
+    private void validateExistMenuGroup(Long menuGroupId) {
+        if (!menuGroupDao.existsById(menuGroupId)) {
             throw new IllegalArgumentException();
         }
+    }
 
+    private MenuProducts generateMenuProducts(MenuCreateRequest menuCreateRequest) {
+        validateExistProduct(menuCreateRequest);
         Map<Long, Long> groupByMenuProductsId = menuCreateRequest.getMenuProducts().stream()
                 .collect(Collectors.toMap(MenuProductsRequest::getProductId, MenuProductsRequest::getQuantity));
 
-        long sum = 0L;
-        final List<MenuProduct> savedMenuProducts = new ArrayList<>();
-        for (Long menuProductsId : groupByMenuProductsId.keySet()) {
-            final Product product = productDao.findById(menuProductsId)
-                    .orElseThrow(IllegalArgumentException::new);
-            sum += product.getPrice() * groupByMenuProductsId.get(menuProductsId);
-            savedMenuProducts.add(new MenuProduct(menuProductsId, groupByMenuProductsId.get(menuProductsId)));
-        }
-
-        if (price.compareTo(sum) > 0) {
-            throw new IllegalArgumentException();
-        }
-
+        List<MenuProduct> savedMenuProducts = new ArrayList<>();
+        groupByMenuProductsId.keySet()
+                .forEach(each -> savedMenuProducts.add(new MenuProduct(each, groupByMenuProductsId.get(each))));
```

Comment:
```suggestion
 List<MenuProduct> savedMenuProducts = groupByMenuProductsId.keySet()
                        .stream()
                        .map(each -> new MenuProduct(each, groupByMenuProductsId.get(each)))
                        .collect(Collectors.toList());
```
[이펙티브 자바 Item46. 스트림에서는 부작용 없는 함수를 사용하라](https://iskull-dev.tistory.com/263?category=998886)를 참고해 주세요